internal/process/provisioning/create_runtime_resource_step.go

```diff
@@ -248,9 +244,55 @@ func (s *CreateRuntimeResourceStep) providerValues(operation *internal.Operation
 	return p.Provide(), nil
 }
 
+func (s *CreateRuntimeResourceStep) createHighAvailabilityConfiguration() *gardener.HighAvailability {
+
+	var failureTolerance *gardener.HighAvailability
+	if s.config.ControlPlaneFailureTolerance == string(gardener.FailureToleranceTypeZone) {
+		failureTolerance = &gardener.HighAvailability{
```

Comment:
Can we shorten this method and only set "Type" field in the if block?

Reply:
It will make it longer.
And which form is easier to read is highly subjective.

Reply:
In the next commit